src/select/select.js

```diff
@@ -43,12 +43,6 @@ class Select extends React.Component<PropsT, StatelessStateT> {
     multiple: false,
     tabIndex: 0,
     textValue: '',
-    filterOption: (option: OptionT, query: string) => {
-      return (
-        typeof option.label === 'string' &&
-        option.label.toLowerCase().indexOf(query.toLowerCase()) >= 0
-      );
-    },
```

Comment:
Moving this out of defaultProps and into a method allows us to utilize `this.getOptionLabel` for our default filtering instead of only filtering on `option.label`.